app/components/app-picker.js

```diff
@@ -1,16 +1,11 @@
 import Component from '@ember/component';
-import { observer } from '@ember/object';
 import { alias, reads } from '@ember/object/computed';
 
 export default Component.extend({
   classNames: ['app-picker'],
 
-  apps: alias('port.detectedApplications.[]'),
-  selectedApp: reads('port.applicationId'),
-
-  selectedDidChange: observer('selectedApp', function() {
-    this.port.set('applicationId', this.selectedApp);
-  }),
```

Comment:
This seems unnecessary, since we already have an action here, and on the port side we also have a callback to maintain the `applicationId`/`applicationName` when they change.